### PR TITLE
WRO-4903: Deprecate moonstone related modules

### PR DESCRIPTION
**Pull request by alexandrumorariu**
### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [X] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Deprecated all the modules that are only used in moonstone to be removed in the future.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRO-4903

### CODE REVIEW

File: packages/ui/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,13 @@ The following is a curated list of changes in the Enact ui module, newest change
 
 ### Deprecated
 
+- `ui/IconButton` to be removed in the future
```
Comment (daniel-stoian-lgp) [inline packages/ui/CHANGELOG.md:15]: let's put "to be removed in 5.0.0" for all these components

File: packages/ui/IconButton/IconButton.js [hunk 2]
```diff
@@ -30,6 +30,7 @@ import componentCss from './IconButton.module.less';
  * @memberof ui/IconButton
  * @ui
  * @public
+ * @deprecated Will be removed soon
```
Comment (daniel-stoian-lgp) [inline packages/ui/IconButton/IconButton.js:33]:
```suggestion
 * @deprecated Will be removed in 5.0.0.
```

Change in all components

File: packages/ui/CHANGELOG.md (hunk 1 above)
Comment (alexandrumorariu) [reply packages/ui/CHANGELOG.md]: Done

File: packages/ui/IconButton/IconButton.js (hunk 2 above)
Comment (alexandrumorariu) [reply packages/ui/IconButton/IconButton.js]: Done

**Review by daniel-stoian-lgp — APPROVED** [review]
LGTM

File: packages/ui/IconButton/IconButton.js [hunk 3]
```diff
@@ -30,6 +30,7 @@ import componentCss from './IconButton.module.less';
  * @memberof ui/IconButton
  * @ui
  * @public
+ * @deprecated Will be removed in 5.0.0.
```
Comment (MikyungKim) [inline packages/ui/IconButton/IconButton.js:33]:
```suggestion
 * @deprecated Will be removed in 5.0.0. Use `ui/Button` instead.
```

File: packages/ui/IconButton/IconButton.js [hunk 4]
```diff
@@ -225,6 +227,7 @@ const IconButtonDecorator = compose(
...
```
Comment (MikyungKim) [inline packages/ui/IconButton/IconButton.js:230]:
```suggestion
 * @deprecated Will be removed in 5.0.0. Use `ui/Button` instead.
```

File: packages/ui/IconButton/IconButton.js (hunk 3 above)
Comment (MikyungKim) [inline packages/ui/IconButton/IconButton.js:33]: Should we add these 3 deprecates? Or can we leave one at line 12?

Comment (MikyungKim) [reply packages/ui/IconButton/IconButton.js:33]: If we can leave one line, Please add `Use `ui/Button` instead.` at the end.

File: packages/ui/RadioDecorator/RadioDecorator.js [hunk 5]
```diff
@@ -58,6 +58,7 @@ const defaultConfig = {
  * @memberof ui/RadioDecorator
  * @hoc
  * @public
+ * @deprecated Will be removed in 5.0.0.
```
Comment (MikyungKim) [inline packages/ui/RadioDecorator/RadioDecorator.js:61]: Here also, Should it be at line 7?

File: packages/ui/Scrollable/ScrollAnimator.js [hunk 6]
```diff
@@ -41,6 +41,7 @@ const
  * @class ScrollAnimator
  * @memberof ui/Scrollable
  * @private
+ * @deprecated Will be removed in 5.0.0.
```
Comment (MikyungKim) [inline packages/ui/Scrollable/ScrollAnimator.js:44]: It is a private module so you don't need to put this.

File: packages/ui/Scrollable/ScrollThumb.js [hunk 7]
```diff
@@ -10,6 +10,7 @@ import css from './ScrollThumb.module.less';
  * @memberof ui/Scrollable
  * @ui
  * @private
+ * @deprecated Will be removed in 5.0.0.
```
Comment (MikyungKim) [inline packages/ui/Scrollable/ScrollThumb.js:13]: It's a private module.

File: packages/ui/Scrollable/Scrollable.js [hunk 8]
```diff
@@ -68,6 +68,7 @@ const TouchableDiv = ForwardRef({prop: 'ref'}, Touchable('div'));
  * @memberof ui/Scrollable
  * @ui
  * @private
+ * @deprecated Will be removed in 5.0.0.
```
Comment (MikyungKim) [inline packages/ui/Scrollable/Scrollable.js:71]: It's a private module.

File: packages/ui/Scrollable/ScrollableNative.js [hunk 9]
```diff
@@ -54,6 +54,7 @@ const TouchableDiv = Touchable('div');
  * @memberof ui/ScrollableNative
  * @ui
  * @private
+ * @deprecated Will be removed in 5.0.0.
```
Comment (MikyungKim) [inline packages/ui/Scrollable/ScrollableNative.js:57]: Here too.

File: packages/ui/Scrollable/Scrollbar.js [hunk 10]
```diff
@@ -34,6 +34,7 @@ const setCSSVariable = (element, variable, value) => {
  * @memberof ui/Scrollable
  * @ui
  * @private
+ * @deprecated Will be removed in 5.0.0.
```
Comment (MikyungKim) [inline packages/ui/Scrollable/Scrollbar.js:37]: Here too.

File: packages/ui/SlotItem/SlotItem.js [hunk 11]
```diff
@@ -27,6 +27,7 @@ import ForwardRef from '../ForwardRef';
  * @memberof ui/SlotItem
  * @ui
  * @public
+ * @deprecated Will be removed in 5.0.0.
```
Comment (MikyungKim) [inline packages/ui/SlotItem/SlotItem.js:30]: Like IconButton, Please check if we could put this line once in this file.

File: packages/ui/ToggleItem/ToggleItem.js [hunk 12]
```diff
@@ -69,6 +69,7 @@ const iconCreator = (position) => ({disabled, icon, iconComponent, iconPosition,
  * @memberof ui/ToggleItem
  * @ui
  * @public
+ * @deprecated Will be removed in 5.0.0.
```
Comment (MikyungKim) [inline packages/ui/ToggleItem/ToggleItem.js:72]: Here too.

File: packages/ui/VirtualList/UiVirtualListBase.js [hunk 13]
```diff
@@ -1181,6 +1182,7 @@ VirtualListBase.displayName = 'ui:VirtualListBase';
  * @memberof ui/VirtualList
  * @ui
  * @private
+ * @deprecated Will be removed in 5.0.0.
```
Comment (MikyungKim) [inline packages/ui/VirtualList/UiVirtualListBase.js:1185]: It's a private module.

File: packages/ui/CHANGELOG.md [hunk 14]
```diff
@@ -2,6 +2,18 @@
 
 The following is a curated list of changes in the Enact ui module, newest changes on the top.
 
+## [unreleased]
+
+### Deprecated
+
+- `ui/IconButton` to be removed in 5.0.0.
```
Comment (MikyungKim) [inline packages/ui/CHANGELOG.md:9]: Please remove private modules from changelog.

File: packages/ui/CHANGELOG.md [hunk 15]
```diff
@@ -2,6 +2,17 @@
...
```
Comment (MikyungKim) [inline packages/ui/CHANGELOG.md:9]:
```suggestion
- `ui/IconButton` to be removed in 5.0.0, use `ui/Button` instead
```

File: packages/ui/CHANGELOG.md [hunk 16]
```diff
@@ -2,6 +2,17 @@
...
+- `ui/RadioDecorator` to be removed in 5.0.0.
+- `ui/Scroller.UiScrollerBase` to be removed in 5.0.0.
+- `ui/SlotItem` to be removed in 5.0.0.
+- `ui/ToggleItem` to be removed in 5.0.0.
+- `ui/VirtualList.UiVirtualListBase` to be removed 5.0.0.
```
Comment (MikyungKim) [inline packages/ui/CHANGELOG.md:14]: Pleaes remove the last `.`.

**Review by MikyungKim — COMMENTED** [review]
Oh, one thing we missed here is we should wrap render function with `deprecate` function from `import deprecate from '@enact/core/internal/deprecate';`.
Please refer to `GridListImageItem`.

**Review by MikyungKim — APPROVED** [review]
LGTM

**Comment by seunghoh** [comment]
Nice work!

**Pull request merged**